pages/markets.page.tsx

```diff
@@ -1,15 +1,139 @@
+import { API_ETH_MOCK_ADDRESS } from '@aave/contract-helpers';
+import { Trans } from '@lingui/macro';
+import { Box, Typography, useMediaQuery, useTheme } from '@mui/material';
+import { useState } from 'react';
+import { ContentContainer } from 'src/components/ContentContainer';
+import { ListWrapper } from 'src/components/lists/ListWrapper';
+import { Link } from 'src/components/primitives/Link';
+import { Warning } from 'src/components/primitives/Warning';
+import { MarketWarning } from 'src/components/transactions/Warnings/MarketWarning';
+import { useAppDataContext } from 'src/hooks/app-data-provider/useAppDataProvider';
+import { useProtocolDataContext } from 'src/hooks/useProtocolDataContext';
 import { MainLayout } from 'src/layouts/MainLayout';
+import { MarketAssetListTitle } from 'src/modules/markets/MarketAssetListTitle';
 import MarketAssetsList from 'src/modules/markets/MarketAssetsList';
 import { MarketsTopPanel } from 'src/modules/markets/MarketsTopPanel';
-
-import { ContentContainer } from '../src/components/ContentContainer';
+import { fetchIconSymbolAndName } from 'src/ui-config/reservePatches';
 
 export default function Markets() {
+  const { reserves, loading } = useAppDataContext();
+  const { currentMarketData, currentNetworkConfig } = useProtocolDataContext();
+  const { breakpoints } = useTheme();
+  const sm = useMediaQuery(breakpoints.down('sm'));
+  const [searchTerm, setSearchTerm] = useState('');
+
+  const filteredData = reserves
+    .filter((res) => res.isActive)
+    .filter((res) => {
+      if (!searchTerm) return true;
+      const term = searchTerm.toLowerCase().trim();
+      return (
+        res.symbol.toLowerCase().includes(term) ||
+        res.name.toLowerCase().includes(term) ||
+        res.underlyingAsset.toLowerCase().includes(term)
+      );
+    })
+    .map((reserve) => ({
+      ...reserve,
+      ...(reserve.isWrappedBaseAsset
+        ? fetchIconSymbolAndName({
+            symbol: currentNetworkConfig.baseAssetSymbol,
+            underlyingAsset: API_ETH_MOCK_ADDRESS.toLowerCase(),
+          })
+        : {}),
+    }));
+
+  const marketFrozen = !reserves.some((reserve) => !reserve.isFrozen);
+  const showFrozenMarketWarning =
```

Comment:
This check isn't necessary since the `MarketWarning` component will just return <></> if the market doesn't have a message there

Reply:
It was reducing two verbose checks into a single statement that could be used twice

Reply:
It’s also human readable / friendly imo

Reply:
The second check on market name isn't necessary though. If a new market warning is added, now it would need to be added to `MarketWarning` and to this array, whereas if the check were removed, it would function the same since the market name check is already handled in `MarketWarning`

Reply:
Resolving - we have discussed doing a refactor of `MarketWarning` and cleaning up all instances of it throughout the app (about 8)

Reply:
Disregard, this will require changes to MarketWarning so it'll be a separate ticket where all of the instances can be updated as well